autoload/slime.vim

```diff
@@ -2,23 +2,54 @@
 " Target Interface
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
+" TargetSend tries to send the text and its config to the right place, by
+" finding the first existing variable in this list and performing the
+" corresponding action.
+" 1. b:slime_target_send  -> Trying to execute a `system` call, passing the
+" configuration in a `CONFIG` environment variable,
+" 2. b:SlimeTargetSend    -> Calling the function with `config` and `text`,
+" 3. g:slime_target_send  ->  Trying to execute a `system` call, passing the
+" configuration in a `CONFIG` environment variable,
+" 4. g:SlimeTargetSend    -> Calling the function with `config` and `text`.
 function! s:TargetSend(config, text)
-  let output = system("CONFIG=" . shellescape(a:config) . " " . g:slime_target_send, a:text)
+  if exists("b:slime_target_send")
+    let output = system("CONFIG=" . shellescape(a:config) . " " . b:slime_target_send, a:text)
+  elseif exists("*b:SlimeTargetSend")||exists("b:SlimeTargetSend") 
+    let output = b:SlimeTargetSend(a:config, a:text)
+  elseif exists("g:slime_target_send")
+    let output = system("CONFIG=" . shellescape(a:config) . " " . g:slime_target_send, a:text)
+  elseif exists("*g:SlimeTargetSend")||exists("g:SlimeTargetSend") 
+    let output = g:SlimeTargetSend(a:config, a:text)
+  else
+    echoerr "vim-slime could not determine a valid target !"
+  endif
```

Comment:
this does feel like too much :thinking: 

like other vim-slime config, I imagined a simpler version where:
- if `b:slime_target_send` exists, use that
- if `g:slime_target_send` exists, snapshot to `b:slime_target_send`, use that
- give up, "missing config" error

why a duality of `slime_target_send` and `SlimeTargetSend`? (that might depend on my vimscript question above)

Reply:
> like other vim-slime config, I imagined a simpler version where:
> 
>     * if `b:slime_target_send` exists, use that
> 
>     * if `g:slime_target_send` exists, snapshot to `b:slime_target_send`, use that
> 
>     * give up, "missing config" error

Yes, this is better.




> why a duality of `slime_target_send` and `SlimeTargetSend`? (that might depend on my vimscript question above)

That's one of vimscript quirks, function names have to start with [a capital letter](https://learnvimscriptthehardway.stevelosh.com/chapters/23.html), so I could not devise a way to have `b:slime_target_send` store either a string to feed `system` or a function for vim to call...

Reply:
maybe my initial thought of "string means a shell command" is restrictive ... what if it always was a function pointer (or a function name, at this point)?